### PR TITLE
Kobo: Yet another attempt at working around the hangs on the latest NXP boards

### PR DESCRIPTION
**Pull request by NiLuJe**
* Notification: Drop the fencing from #10083; it never actually helped, and had subtle side-effects we could do without.
* VirtualKeyBoard: Flash on close, otherwise, some of the fast refresh glitches may be burned into the working buffer until a flash. Making sure we flash ourselves prevent it from sticking around on the page ;).
* util: Move `writeToSysfs` to base (i.e., `ffi/util`), as we need it there (and it actually makes more sense there anyway ;p).
* Bump base for https://github.com/koreader/koreader-base/pull/1645, which is where the actual workaround (hopefully) lives.

Re #8414, #9806, #10558

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10771)

### CODE REVIEW

**Comment by NiLuJe** [comment]
Oh, and this also re-enables the jump marker on affected devices ;).

Sadly, as described in the comments and the base PR, this didn't help with the wait_for ioctl timeouts, so we've kept our crappy workaround. If you do manually disable it, you do get a slightly different behavior, in that it'll now (again, on affected devices) do those ioctls in the same weird pairs as Nickel.

FWIW, it took me over 2k page turns to get a couple of timeouts (back to back, things start to go wonky quick after the first one).... And then, like, 15 page turns in Nickel after a reboot, lol >_<".
Anyway, I'm a bit miffed about losing the fences offered by the proper implementation here, but, thankfully (except for the Nia, but the Nia basically shouldn't exist anyway), the affected devices have screens that are fast enough that it's not too much of a problem, besides some slightly weirder looking refreshes in some contexts, if you know what to look for ;).

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/10771)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @Frenzie, @pazos, and @poire-z)

**Review by Frenzie — APPROVED** [review]

**Comment by srvoets** [comment]
> Oh, and this also re-enables the jump marker on affected devices ;).
> 
> Sadly, as described in the comments and the base PR, this didn't help with the wait_for ioctl timeouts, so we've kept our crappy workaround. If you do manually disable it, you will get a slightly different behavior than before, in that it'll now (again, on affected devices) do those ioctls in the same weird pairs as Nickel.
> 
> FWIW, it took me over 2k page turns to get a couple of timeouts (back to back, things start to go wonky quick after the first one).... And then, like, 15 page turns in Nickel after a reboot, lol >_<". Anyway, I'm a bit miffed about losing the fences offered by the proper implementation here, but, thankfully (except for the Nia, but the Nia basically shouldn't exist anyway), the affected devices have screens that are fast enough that it's not too much of a problem, besides some slightly weirder looking refreshes in some contexts, if you know what to look for ;).


Thank you so much, I just got a kobo a few weeks ago and put koreader on immediately, never touched the stock FW.

But after the freezing I attempted stock (patched) FW, but I still missed koreader.


How do I get this update on my device like yesterday?

**Comment by NiLuJe** [comment]
If you have an existing installation, just switch to the Dev channel and do an OTA update ;).

Otherwise, you can grab a current nighly build: https://build.koreader.rocks/download/nightly/

**Comment by srvoets** [comment]
> If you have an existing installation, just switch to the Dev channel and do an OTA update ;).
> 
> Otherwise, you can grab a current nighly build: https://build.koreader.rocks/download/nightly/

I wanted to come back and say that I did this about 3 weeks ago, and in that time I only had 2 freezes, which only required me to tap the power button once to put it to sleep, and then I was able to wake it, and start reading.

contrast this with the previous behavoir which had me doing a ahrd reset almost daily.

So thank you!

I am currently still on dev branch, has this fix been merged to the normal release?

Basically, can I switch my branch back to stable now, was this fix implemented into the stable release?

**Comment by poire-z** [comment]
> in that time I only had 2 freezes, which only required me to tap the power button once to put it to sleep, and then I was able to wake it, and start reading.

On Kobo GloHD, this is supposed to be an issue with autostandby, cf #10133 - no really a freeze, but a touch panel issue (as the power button does work).
Do you happen to have that plugin (Auto suspend, which also manages auto standby) enabled and active ? If yes, and if you disable auto-standby, does this faux-freeze disappear?

**Comment by NiLuJe** [comment]
On which device? IIRC, none of the affected devices have an IR grid, which I think was the culprit on the Glo HD, so that's... mildly weird.

(I mean, autostandby *can* appear to send a few input events to the void on some devices, so it's not entirely out of question; I *definitely* didn't test autostandby thoroughly because it doesn't make much sense on most of these devices).

**Comment by NiLuJe** [comment]
Re: https://github.com/koreader/koreader/issues/10558

The Nia does *NOT* actually provide the necessary sysfs knob (at least according to the published kernel sources), so, that's yet another strike against it making any kind of sense as a viable device ;o).
